tests/pexpects/commandline.py

```diff
@@ -12,6 +13,92 @@
 )
 expect_prompt()
 
+# Setup commandline test bindings
+sendline(r"""function query_commandline;\
+                 printf '\n%s ' (commandline --cursor); commandline;\
+                 commandline -f cancel-commandline;\
+             end;\
+             bind \eQ query_commandline;\
+             bind \eT test_commandline;""")
+expect_prompt()
+
+# Run a commandline test
+#   commandline is the command line text
+#   binding is the function body for test_commandline
+#   expect is the expected value of $result (see query_commandline above)
+def t(commandline, binding, expect):
+    sendline(f"function test_commandline; {binding}; end")
+    expect_prompt()
+    sendline(f"{commandline}\x1bT\x1bQ")
+    expect_prompt(escape(f"\r\n{expect}\r\n"))
+
+# commandline --cursor
+t("abcdef", "commandline --cursor 3", "3 abcdef")
+t("abc defghi jkl",
+  "commandline --cursor 8; commandline --current-token --cursor 1",
+  "5 abc defghi jkl")
+t("abc def", "commandline --current-token --cursor -- -3", "1 abc def")
+t("abc def", "commandline --current-token --cursor -- -100", "0 abc def")
+t("abc def",
+  "commandline --current-token --cursor 0; commandline --insert x",
+  "5 abc xdef")
+t("echo ===; echo hello | echo there",
+  "commandline --current-job --cursor 5",
+  "14 echo ===; echo hello | echo there" )
+t("echo hello | echo there",
+  "commandline --current-process --cursor 4",
+  "16 echo hello | echo there")
+
+# commandline --insert
+t("abc", "commandline --cursor 2; commandline --insert x", "3 abxc")
+t("abc defghi jkl",
+  "commandline --cursor 8; commandline --current-token --insert x",
```

Comment:
Okay, this test shows that `--current-token` is ignored when `--insert` is given